ddtrace/tracer/tracer.go

```diff
@@ -206,13 +212,13 @@ func (t *tracer) worker(tick <-chan time.Time) {
 			for {
 				select {
 				case trace := <-t.payloadChan:
-					t.pushPayload(trace)
+					t.traceHandler.pushPayload(trace)
 				default:
 					break loop
 				}
 			}
 			t.config.statsd.Incr("datadog.tracer.flush_triggered", []string{"reason:shutdown"}, 1)
-			t.flush()
+			t.traceHandler.flush()
```

Comment:
I guess this should technically not be needed give that we would call `t.traceWriter.stop` or similar in `Stop` right?

Reply:
Well, if each `traceWriter` has it's own tick function (essentially the worker function), we should remove the worker function from the tracer and give it to the `traceWriter`s